src/nextjournal/clerk/viewer.cljc

```diff
@@ -1349,26 +1352,29 @@
 
 (declare assign-closing-parens)
 
-(defn process-render-fn [{:as viewer :keys [render-fn evaluator]}]
+(defn process-render-fn [{:as viewer :keys [render-fn render-evaluator]}]
   (cond-> viewer
     (and render-fn (not (viewer-fn? render-fn)))
     (update :render-fn (fn [rf]
                          (assoc (->viewer-fn rf)
-                                :evaluator (or evaluator :sci))))))
+                                :render-evaluator (or render-evaluator :sci))))))
 
 (defn hash-sha1 [x]
   #?(:clj (analyzer/valuehash :sha1 x)
      :cljs (let [hasher (goog.crypt.Sha1.)]
              (.update hasher (goog.crypt/stringToUtf8ByteArray (pr-str x)))
              (.digest hasher))))
 
-(defn process-viewer [viewer]
+(defn process-viewer [viewer {:nextjournal/keys [render-evaluator]}]
   (if-not (map? viewer)
     viewer
-    (-> viewer
-        (dissoc :pred :transform-fn :update-viewers-fn)
-        (as-> viewer (assoc viewer :hash (hash-sha1 viewer)))
-        (process-render-fn))))
+    (let [evaluator (:render-evaluator viewer)]
```

Comment:
Maybe drop this let? you get a cleaner diff :-) I also checked `process-viewer` is called in just the one spot below, so this arity change should be fine.